src/NServiceBus.Core/Unicast/MessageOperations.cs

```diff
@@ -17,7 +17,10 @@ public static Task Publish<T>(IBehaviorContext context, Action<T> messageConstru
 
         public static Task Publish(IBehaviorContext context, object message, PublishOptions options)
         {
-            return Publish(context, message.GetType(), message, options);
+            var mapper = context.Builder.Build<IMessageMapper>();
```

Comment:
could we reuse the mapper instead of building it every time?

Reply:
Given that the mapper is registered as a singleton 

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Serialization/SerializationFeature.cs#L58

I'm not sure if it's worth the extra code we would have to add? (since this class is static we would need to pass it to all methods?)

Reply:
Talked to @bording and agreed to add a static field to cache the mapper